dataland-frontend/src/components/forms/CreateEuTaxonomyForFinancials.vue

```diff
@@ -554,14 +580,37 @@ export default defineComponent({
       try {
         this.postEuTaxonomyDataForFinancialsProcessed = false;
         this.messageCount++;
+        
+        // JSON.parse/stringify used to clone the formInputsModel in order to stop Proxy refreneces
+        const clonedFormInputsModel = JSON.parse(JSON.stringify(this.formInputsModel)) as ObjectType;
+        const kpiSectionsModel = (clonedFormInputsModel.data as ObjectType).kpiSectionsModel;
+
+        const eligibilityKpis = Object.keys(kpiSectionsModel)
+          .map( key => ({ key, fieldName: euTaxonomyKPIsModel.kpisFieldNameToFinancialServiceType[key] as string }) )
+          .map( kpi => ({ [kpi.fieldName]: kpiSectionsModel[kpi.key][kpi.fieldName] }) )
```

Comment:
if `kpi.fieldName` is `undefined` this property just does not get set?

Reply:
We're running this map only on the kpis found in `kpiSectionsModel` . There's no chance it would be undefined as they are previously created from `euTaxonomyKPIsModel.kpisFieldNameToFinancialServiceType`

Reply:
Ok. I find that somewhat hard to notice when you look at the code. Could we emphasize that by putting the generation of the object into a function. Or is this too much overhead?